### PR TITLE
Initial recipe for FreeIPMI version 1.5.1

### PR DESCRIPTION
**Pull request by jimklimov**
Essentially a port and update of old freeipmi-0.7.7 integration recipe from OpenSolaris - in the end, they happened to use vanilla freeipmi + manpage fixes (dropped here) and SMF instead of init-scripts. So here that went too. Says it has sunbmc among other devices ;)

### CODE REVIEW

**Comment by jimklimov** [comment]
Update of the package from old 0.7.7 to this (in private repo): 

```
$ sudo pkg install -v pkg://oi-userland-jim/system/management/freeipmi
            Packages to update:         1
            Services to change:         2
     Estimated space available:   3.35 GB
Estimated space to be consumed: 367.37 MB
       Create boot environment:        No
Create backup boot environment:       Yes
          Rebuild boot archive:        No

Changed packages:
openindiana.org -> oi-userland-jim
  system/management/freeipmi
    0.7.7-2013.0.0.0 -> 1.5.1-2016.0.0.0

Services:
  restart_fmri:
    svc:/application/texinfo-update:default
    svc:/system/manifest-import:default

Editable files to change:
  Remove:
    etc/freeipmi.conf
    etc/ipmi_monitoring_sensors.conf
  Install:
    etc/freeipmi/freeipmi.conf
    etc/freeipmi/freeipmi_interpret_sel.conf
    etc/freeipmi/freeipmi_interpret_sensor.conf
    etc/freeipmi/ipmidetect.conf
    etc/freeipmi/ipmidetectd.conf
    etc/freeipmi/ipmiseld.conf
    etc/freeipmi/libipmiconsole.conf
  Update:
    var/lib/freeipmi/ipckey

DOWNLOAD                                PKGS         FILES    XFER (MB)   SPEED
Completed                                1/1       415/415      7.5/7.5    0B/s

PHASE                                          ITEMS
Removing old actions                           88/88
Installing new actions                       321/321
Updating modified actions                    153/153
Updating package state database                 Done 
Updating package cache                           1/1 
Updating image state                            Done 
Creating fast lookup database                   Done 
Updating package cache                           6/6 
```

**Comment by xen0l** [comment]
Changes look good so far. How did you test this?

**Comment by jimklimov** [comment]
So far only did compilation of freeipmi itself, and its compatibility with requirements of NUT. I do not have Hipster servers, but can try a build on OI servers to see if their SunBMCs are seen and Supermicro IPMIs can be contacted with this freeipmi version, etc...

**Comment by pyhalov** [comment]
Can we somehow go on with this?..

**Comment by jimklimov** [comment]
Merge it? Announce on mailing list that testing against ipmi/bmc hardware
is needed and people can install the package from your :10000 (and how) and
that in say a week you'll merge?

## 

Typos courtesy of GMail on my Samsung Android
31.05.2016 22:41 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by pyhalov** [comment]
Any instructions on testing?

**Comment by jimklimov** [comment]
I had a couple of plans but did not yet get around to do them:
1) look at some ipmi card with ipmitool (request status, etc.)
2) on a server with compatible bmc, run a bmc-watchdog

These are things that worked with opensolaris/sol10 freeipmi which is
vanilla 0.7.7 + smf scripts.

## 

Typos courtesy of GMail on my Samsung Android
01.06.2016 7:40 пользователь "Alexander Pyhalov" notifications@github.com
написал:

File: components/sysutils/freeipmi/Solaris/svc-bmc-watchdog [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		echo "svc-bmc-watchdog: bmc-watchdog is disabled"
+	fi
+else
+	case "$1" in
+	'start')
+		# bmc-watchdog requires the presence of a BMC character device
+		# to run.  If one is not detected, then disable the service
+		# and exit.
+
+		if [ ! -c /dev/bmc ]; then
```
Comment (pyhalov) [inline components/sysutils/freeipmi/Solaris/svc-bmc-watchdog]: On illumos it should be /dev/ipmi0

File: components/sysutils/freeipmi/Solaris/svc-ipmidetectd [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		echo "svc-ipmidetectd: ipmidetectd is disabled"
+	fi
+else
+	case "$1" in
+	'start')
+		# ipmidetectd requires the presence of a BMC character device
+		# to run.  If one is not detected, then disable the service
+		# and exit.
+
+		if [ ! -c /dev/bmc ]; then
```
Comment (pyhalov) [inline components/sysutils/freeipmi/Solaris/svc-ipmidetectd]: ipmidetectd can work over network, this check is not necessary

File: components/sysutils/freeipmi/Solaris/svc-ipmiseld [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		echo "svc-ipmiseld: ipmiseld is disabled"
+	fi
+else
+	case "$1" in
+	'start')
+		# ipmiseld requires the presence of a BMC character device
+		# to run.  If one is not detected, then disable the service
+		# and exit.
+
+		if [ ! -c /dev/bmc ]; then
```
Comment (pyhalov) [inline components/sysutils/freeipmi/Solaris/svc-ipmiseld]: ipmiseld also can work over network

**Comment by pyhalov** [comment]
Please, change check for /dev/bmc to check for /dev/ipmi0 for watchdog service, remove checks for /dev/bmc in other manifests and let's push this

**Comment by jimklimov** [comment]
Ok. Is there also a pkg-fmri to depend on that would provide the ipmi0
driver? Do we want to require one - not all systems have a bmc or manage
themselves, or are GZs ;) - but if the driver is small, perhaps makes sense
to pull it? Or make an explicit bmc-watchdog fmri that provides the smf
service and requires both the freeipmi and driver packages.

## 

Typos courtesy of GMail on my Samsung Android
17.06.2016 8:04 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by pyhalov** [comment]
There is driver/ipmi , but we don't want to depend on it, perhaps people just use ipmi over lan...

**Comment by pyhalov** [comment]
Seems to be ready. Please, merge commits.

**Comment by jimklimov** [comment]
Comments addressed, bump :)

**Pull request merged**